src/platform/unix/mod.rs

```diff
@@ -472,7 +472,12 @@ impl OsIpcReceiverSet {
         let mut selection_results = Vec::new();
         match self.poll.poll(&mut self.events, None) {
             Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
+            Err(ref e) if e.kind() != ErrorKind::Interrupted => {
```

Comment:
Good catch. I changed the control flow right before committing and forgot to change this.